h2o-webserver-iface/src/main/java/water/webserver/iface/H2OHttpView.java

```diff
@@ -0,0 +1,38 @@
+package water.webserver.iface;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+
+/**
+ * Exposes part of H2O functionality for the purposes of HTTP server adapter.
+ * Contains also logic for handling authentication and other functionality, so that it can be easily shared among various server implementations.
+ */
+public interface H2OHttpView {
+
+  /**
+   * @return configuration related to HTTP server
+   */
+  H2OHttpConfig getConfig();
+
+  /**
+   * @return map of servlets with their context paths
+   */
+  LinkedHashMap<String, Class<? extends HttpServlet>> getServlets();
+
+  /**
+   * @return custom authentication extensions if any
+   */
+  Collection<RequestAuthExtension> getAuthExtensions();
```

Comment:
- Can we guarantee a `Set`, instead of general `Collection` ? There is a HashMap used in ExtensionManager, should be trivial to obtain EntrySet.
- Let's specify if the returned value is a copy of the original collection/set or a direct reference to the original. I'd return a copy only in case the original instance is ImmutableHashmap (from Google, available in ExtensionManager).

Reply:
`Set` - could be, not sure if it is better;
`Map` and `EntrySet` - what would that be good for? I don't need tuples, just a collection of auth extensions, which is in reality mostly empty, and sometimes (for a customer) it has a single member in it... so I would even consider generalizing to just `Iterable`, WDYT?